src/core/scene/a-scene.js

```diff
@@ -58,6 +58,7 @@ class AScene extends AEntity {
     self.time = self.delta = 0;
 
     self.behaviors = {tick: [], tock: []};
+    self.behaviorIterators = {tick: -1, tock: -1};
```

Comment:
not sure if iterator is the best name here. looked at the code for 5 mins without context and hard to understand

Reply:
Open for suggestions. The name seemed quite fitting to me as an iterator over an array is effectively just an index and it makes it clear that it's relevant when we're iterating over the behaviours.

Perhaps `behaviorCursors` works better? It places a bit more emphasis on pointing to a specific element, which is also the invariant that should be upheld when adding or removing behaviours mid-iteration.

Reply:
There's are JS Iterators[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator] so not sure